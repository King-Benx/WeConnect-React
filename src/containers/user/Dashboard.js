import React from 'react';
import business_image from '../../images/business.png';
import DashboardNavigation from '../../navigation/DashboardNavigation';
import { Image } from 'react-bootstrap'

class Dashboard extends React.Component{
    render(){
        return (  
            <div className="row"> 
                <DashboardNavigation/>
                <div className="col-sm-9 content-wrapper">
                   <Image src={ business_image } alt="" rounded responsive/>
                </div>
            </div>
        );
    }
}
export default Dashboard;