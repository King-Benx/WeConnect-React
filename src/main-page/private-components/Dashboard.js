import React from 'react';
import {Redirect} from 'react-router-dom';
import business_image from '../img/business.png';
import DashboardNavigation from '../shared-components/DashboardNavigation';
import CustomFunctions from '../functional-resources/CustomFunctions';

class Dashboard extends React.Component{
    render(){
        return (
            CustomFunctions.checkAuthentication() ? (   
            <div className="row"> 
                <DashboardNavigation/>
                <div className="col-sm-9 content-wrapper">
                   <img src={business_image} className="img img-responsive" alt=""/>
                </div>
            </div>
            ): <Redirect to={{pathname:'/login'}}/>
        );
    }
}
export default Dashboard;