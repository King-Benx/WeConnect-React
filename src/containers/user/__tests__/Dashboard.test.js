import React from 'react';
import { shallow } from 'enzyme';
import Dashboard from "../Dashboard";
import renderer from 'react-test-renderer';

describe('These are tests for the login form ', () =>{

    
    let mountedDashboard;

    beforeEach(()=>{
        mountedDashboard= shallow(<Dashboard />);
    });

    it('renders without crashing', () => {
        shallow(<Dashboard />);
        const tree = renderer.create(<Dashboard />).toJSON();
        expect(tree).toMatchSnapshot();
    });
    
    it('has divs', () => {
        const divs = mountedDashboard.find('div');
        expect(divs.length).toBe(2)
    });

    it('has a dashboard Navigation', () => {
        const dashboardNavigations = mountedDashboard.find('DashboardNavigation');
        expect(dashboardNavigations.length).toBe(1);
    });

    it('has an image', () => {
        const images = mountedDashboard.find('Image');
        expect(images.length).toBe(1)
    });
});