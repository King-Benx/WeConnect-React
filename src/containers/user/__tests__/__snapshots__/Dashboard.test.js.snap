// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`These are tests for the login form  renders without crashing 1`] = `
<div
  className="row"
>
  <div
    className="col-sm-9 content-wrapper"
  >
    <img
      alt=""
      className="img-responsive img-rounded"
      src="business.png"
    />
  </div>
</div>
`;
